internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -317,6 +318,10 @@ func validatingToWebhookDescription(webhook admissionregv1.ValidatingWebhook, de
 	if serviceRef := webhook.ClientConfig.Service; serviceRef != nil {
 		if serviceRef.Port != nil {
 			description.ContainerPort = *serviceRef.Port
+			description.TargetPort = &intstr.IntOrString{
```

Comment:
@estroz : With this change we are defaulting container and target port to the same value specified in service reference. If we do so, how would the user be able to change only the targetPort value? Also, if the service reference is nil, the scaffolding would still not contain `containerPort` / `targetPort`. Considering this, just wanted to confirm if this is the right change ?

Reply:
This value is correct (the service reference should contain the target port of the webhook service). However container port is not, and should be derived from the webhook-associated service. You can modify [this function](https://github.com/operator-framework/operator-sdk/blob/339b4c9c4c20bbf471dd4c0edf16c472d7d4c24a/internal/generate/clusterserviceversion/clusterserviceversion_updaters.go#L368) to return the full service object, from which you can get both container and target ports. If either are empty, let OLM default them.